src/mcnp_library.cpp

```diff
@@ -126,67 +126,34 @@ MCNPLibrary::MCNPLibrary(const std::string& fname) : NDLibrary(fname) {
   std::string num_entries_str;
   std::string temp_str;
   std::string ptable_str;
-  directory_stream >> zaid_str;  // Get rid of 'directory' in stream
-  bool get_zaid_str = true;
-  while (directory_stream.eof() == false) {
-    if (get_zaid_str) {
-      directory_stream >> zaid_str;
-    } else {
-      zaid_str = zaid_str_new;
-    }
-
-    directory_stream >> awr_str;
-    directory_stream >> fname_str;
-
-    directory_stream >> access_str;
-    if (access_str == "+") {
-      directory_stream >> access_str;
-    }
-
-    directory_stream >> ftype_str;
-    if (ftype_str == "+") {
-      directory_stream >> ftype_str;
-    }
-
-    directory_stream >> address_str;
-    if (address_str == "+") {
-      directory_stream >> address_str;
-    }
-
-    directory_stream >> tab_len_str;
-    if (tab_len_str == "+") {
-      directory_stream >> tab_len_str;
-    }
-
-    directory_stream >> record_len_str;
-    if (record_len_str == "+") {
-      directory_stream >> record_len_str;
-    }
-
-    directory_stream >> num_entries_str;
-    if (num_entries_str == "+") {
-      directory_stream >> num_entries_str;
-    }
-
-    directory_stream >> temp_str;
-    if (temp_str == "+") {
-      directory_stream >> temp_str;
-    }
-
-    directory_stream >> ptable_str;
-    if (ptable_str == "+") {
-      // The "ptable" keyword is on the next line. Go grab it.
-      directory_stream >> ptable_str;
-      get_zaid_str = true;
-    } else if (ptable_str == "ptable") {
-      // We just read "ptable" and it was on the main line
-      get_zaid_str = true;
-    } else {
-      // The next item wasn't "+" or "ptable", so we actually just read the next
-      // ZAID. Keep it for later.
-      zaid_str_new = ptable_str;
-      get_zaid_str = false;
+  std::vector<std::string> line_buffer;
+  std::string str, line;
+  std::getline(directory_stream, line); // Get rid of 'directory' in stream
+  while (directory_stream.eof() == false ) {
+    LINE_LOOP:
+    std::getline(directory_stream, line); // get one line
+    // remove leading, trailing and extra spaces in the line:
+    // by Evgeny Karpov in https://shorturl.at/biH35
+    line = std::regex_replace(line, std::regex("^ +| +$|( ) +"), "$1");
+    std::stringstream linestream(line);
+    //linestream.ignore();
+    while ( std::getline(linestream, str, ' ') ) { // split the line and store
+      if (str!="+" && line_buffer.size()<=11)      // each content if it's not
+        line_buffer.push_back(str);                // the continuation mark
+      else
+        goto LINE_LOOP;                            // goto the continued line
```

Comment:
Would it be possible to remove this `goto` ? I think a good solution might be to put the 3 lines above into a lambda function, and we can then call that at the beginning of the outer while loop, and in this else statement.